LoveMachine.Core/ButtplugController.cs

```diff
@@ -192,6 +220,23 @@ protected void DoVibrate(float intensity, int girlIndex, Bone bone)
             client.VibrateCmd(intensity, girlIndex, bone);
         }
 
+        protected internal IEnumerator DoRotate(int girlIndex, Bone bone,
+                    float strokeTimeSecs)
+        {
+
+            float speed = Mathf.Lerp(0.3f, 1f, 0.4f / strokeTimeSecs) * CoreConfig.RotationSpeedRatio.Value;
+            client.RotateCmd(speed: speed, clockwise: clockwise, girlIndex, bone);
+            yield return new WaitForSecondsRealtime(strokeTimeSecs - 0.01f);
+            client.RotateCmd(speed: 0, clockwise: clockwise, girlIndex, bone);
+            // apply sleep to avoid sending speed 0 and next rotation at the same time
+            yield return new WaitForSecondsRealtime(0.01f);

```

Comment:
Okay but why stop the rotation for just one frame?

Reply:
For some reason, rotators (at least A10 cyclone) can't handle same-ish timing command well.
Let's say we have the following code:

```cs
client.RotateCmd(speed: 0, clockwise: clockwise, girlIndex, bone);
client.RotateCmd(speed: 1, clockwise: clockwise, girlIndex, bone);
```

We expect it rotates in full speed but actually it randomly stops forever.

However it seems having sleep works anytime:

```cs
client.RotateCmd(speed: 0, clockwise: clockwise, girlIndex, bone);
yield return new WaitForSecondsRealtime(0.01f);
client.RotateCmd(speed: 1, clockwise: clockwise, girlIndex, bone);
```

I guess rotators confuse commands order if multiple commands are sent in the same time.

That's why we need sleep here.

Reply:
I mean why stop it in the first place if it's only for one frame? Why not just leave it on? What difference does it make if stops for one frame or keeps moving for one frame?

Reply:
Nope, OnEndH is called when exiting H. It should be stopped when IsIdle is true.

Reply:
@Sauceke I've noticed sometime rotator doesn't stop after H.
I've confirmed this behavior on KKS.
I guess sometimes OnEndH and/or IsIdle doesn't work well for vibrator as well.
Do you have any idea?

Reply:
Oh yes, it actually has to be stopped in both places. Sorry for the confusion.
When an H scene ends, all coroutines are immediately stopped, so we have to clean up after them, including resetting all device states. This is what OnEndH() does.
So if you add a loop for stopping rotators right below the one that stops vibrators, it should solve the problem.